src/utils/api/calls.js

```diff
@@ -178,3 +150,50 @@ exports.getTutors = async function getTutors() {
 
 	return { tutors };
 };
+
+async function getAccessToken() {
+	const cacheContent = getCache(key);
+	if (cacheContent) {
+		logger.info('Fetched Access token from cache');
+		return cacheContent.token;
+	}
+	const url = `${COUGARCS_CLOUD_URL}/login`;
+	const data = { COUGARCS_CLOUD_ACCESS_KEY, COUGARCS_CLOUD_SECRET_KEY };
+	logger.info('Fetching access token.');
+	const res = await axios.post(url, data);
+	logger.info('Stored token in cache');
+	setCache(key, { token: res.data.token }, CACHE_TIME);
+	return res.data.token;
+}
+
+exports.postContact = async function postContact({
+	transaction,
+	uhID,
+	email,
+	firstName,
+	lastName,
+	phone,
+	shirtsize,
+	membershipStart,
+	paidUntil,
+}) {
+	const token = getAccessToken();
+
+	const data = {
```

Comment:
This data object is invalid. CougarCS Cloud postContact expects:
```
"membershipStart": {"Year": "YYYY", "Term": "<TERM>"},
"membershipEnd": {"Year": "YYYY", "Term": "<TERM>"}
```
where <TERM> = "Spring" || "Summer" || "Fall" || "Winter"

These two parameters will need to be calculated from the available information. Here are the term ranges as defined in CougarCS Cloud:
```
const TERMS = { Spring: 'May 13', Summer: 'August 23', Fall: 'December 16', Winter: 'January 12' }
```

Reply:
> This data object is invalid. CougarCS Cloud postContact expects:
> 
> ```
> "membershipStart": {"Year": "YYYY", "Term": "<TERM>"},
> "membershipEnd": {"Year": "YYYY", "Term": "<TERM>"}
> ```
> 
> where <TERM> = "Spring" || "Summer" || "Fall" || "Winter"
> 
> These two parameters will need to be calculated from the available information. Here are the term end dates as defined in CougarCS Cloud:
> 
> ```
> const TERMS = { Spring: 'May 13', Summer: 'August 23', Fall: 'December 16', Winter: 'January 12' }
> ```
> 
> paidUntil is ignored

If the user applies after May 13 and before August 23 it will be the Spring term?

Reply:
> If the user applies after May 13 and before August 23 it will be the Spring term?

To answer your question, that would be considered Summer term.

However, because we are designing the code which makes this decision, we have the freedom to decide what term/year would be the start and calculate the ending term/year. In my opinion, summer enrollments should automatically round up to Fall for the start term/year, and winter enrollments should automatically round up to Spring for start.

Reply:
Something like:
REGISTRATION_DATE = today
if REGISTRATION_DATE < May 13: membershipStart.Term = "Spring"
else if REGISTRATION_DATE < December 16: membershipStart.Term = "Fall"

Reply:
So the it will look like this mostly: 

`const TERMS = { Spring: 'May 13',  Fall: 'December 16'}
`

Reply:
@crudemandarin  I have scenarios let me know if the data is right: 

## User register to be a member for a ***semester*** on January 12th 2021: 
```json
{
"membershipStart": {"Year": "2021", "Term": "Spring"},
"membershipEnd": {"Year": "2021", "Term": "Spring" }
}
```

## User register to be a member for a ***year*** on January 12th 2021:
```json
{
"membershipStart": {"Year": "2021", "Term": "Spring"},
"membershipEnd": {"Year": "2021", "Term": "Fall" }
}
```

## User register to be a member for a ***semester*** on May 13th 2021 (This date is based on given end of Spring):
```json
{
"membershipStart": {"Year": "2021", "Term": "Fall"},
"membershipEnd": {"Year": "2021", "Term": "Fall" }
}
```

## User register to be a member for a ***year*** on May 13th 2021 (This date is based on given end of Spring): 
```json
{
"membershipStart": {"Year": "2021", "Term": "Fall"},
"membershipEnd": {"Year": "2022", "Term": "Spring" }
}
```

## User register to be a member for a ***semester*** on July 23th 2021:
```json
{
"membershipStart": {"Year": "2021", "Term": "Fall"},
"membershipEnd": {"Year": "2021", "Term": "Fall" }
}
```

## User register to be a member for a ***year*** on July 23th 2021:
```json
{
"membershipStart": {"Year": "2021", "Term": "Fall"},
"membershipEnd": {"Year": "2022", "Term": "Spring" }
}
```

## User register to be a member for a ***semester*** on December 16th 2021 (This date is based on given end of Fall):
```json
{
"membershipStart": {"Year": "2022", "Term": "Spring"},
"membershipEnd": {"Year": "2022", "Term": "Spring" }
}
```

## User register to be a member for a ***year*** on December 16th 2021 (This date is based on given end of Fall):
```json
{
"membershipStart": {"Year": "2022", "Term": "Spring"},
"membershipEnd": {"Year": "2022", "Term": "Fall" }
}
```

## User register to be a member for a ***semester*** on December 21th 2021:
```json
{
"membershipStart": {"Year": "2022", "Term": "Spring"},
"membershipEnd": {"Year": "2022", "Term": "Spring" }
}
```

## User register to be a member for a ***year*** on December 21th 2021:
```json
{
"membershipStart": {"Year": "2022", "Term": "Spring"},
"membershipEnd": {"Year": "2022", "Term": "Fall" }
}
```